<a href="https://colab.research.google.com/github/ardhiansyahprimadikaadi/AI_Praktikum-1_09020624024/blob/main/AI_Praktikum_Linear_Logistik.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Praktikum Analisis Code Regresi Linear Logistik**
## Dataset : Keputusan Pembelian Mobil
### Nama : Ardhiansyah Primadika Adi
### Kelas : H6C
### NIM : 09020624024


---

### **Langkah 1 : Impor Library**
Tahap awal adalah memanggil semua modul yang diperlukan untuk pengolahan data.
*   pandas: Berfungsi untuk mengelola data dalam bentuk tabel agar mudah diproses.
*   train_test_split: Berfungsi membagi data menjadi bagian latihan dan bagian ujian guna validasi model.
*   LogisticRegression: Algoritma utama yang digunakan untuk mengklasifikasikan data ke dalam kategori (seperti Ya/Tidak).
*   Metrics (Accuracy, Confusion Matrix): Alat ukur untuk mengevaluasi ketepatan tebakan model dan melihat letak kesalahan prediksi.





In [1]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report

### **Langkah 2: Menyiapkan Dataset**
Variabel X adalah Usia dan Gaji, sedangkan Y adalah keputusan (1 = Beli, 0 = Tidak Beli).

In [2]:
data_logistik = {
    'usia': [22, 25, 47, 52, 46, 56, 27, 29, 32, 40, 50, 59, 21, 35, 20],
    'gaji_jutaan': [15, 20, 80, 110, 90, 150, 30, 45, 50, 65, 85, 120, 18, 40, 15],
    'beli_mobil': [0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 1, 1, 0, 0, 0]
}
df_l = pd.DataFrame(data_logistik)
print(df_l)

    usia  gaji_jutaan  beli_mobil
0     22           15           0
1     25           20           0
2     47           80           1
3     52          110           1
4     46           90           1
5     56          150           1
6     27           30           0
7     29           45           0
8     32           50           0
9     40           65           0
10    50           85           1
11    59          120           1
12    21           18           0
13    35           40           0
14    20           15           0


### Analisis
Dataset ini mencerminkan pola perilaku konsumen di mana faktor usia yang lebih matang dan pendapatan yang lebih tinggi biasanya berkolerasi positif dengan keputusan untuk melakukan pembelian barang mewah seperti mobil.

### **Langkah 3: Menentukan Variabel X dan y**
Menentukan mana faktor penyebab (X) yaitu usia dan gaji jutaan dan mana hasil yang ingin ditebak (Y) yaitu beli mobil.

In [3]:
X = df_l[['usia', 'gaji_jutaan']]
y = df_l['beli_mobil']

### Analisis
Penentuan variabel ini bertujuan untuk melihat bagaimana dua faktor demografi dan ekonomi tersebut secara bersama-sama menentukan probabilitas seseorang masuk ke dalam kategori Pembel atauBukan Pembeli.

### **Langkah 4: Pembagian Data**
Data dibagi agar model dapat diuji kemampuannya dalam memprediksi keputusan orang baru yang profil usia dan gajinya tidak ada dalam data latihan.

In [4]:
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42
)

### Analisis
Pembagian data dilakukan untuk menguji validitas model. Penggunaan data uji (testing set) berfungsi untuk memastikan model tidak sekadar menghafal data, melainkan mampu mengenali pola perilaku konsumen secara umum.


### **Langkah 5: Inisialisasi dan Pelatihan Model**
Pada tahap ini, model mempelajari pola data dimana nodel dibuat menggunakan objek LinearRegression() dan dilatih dengan fungsi .fit() untuk mempelajari variabel input (usia & gaji) dengan target (keputusan beli).

In [5]:
model = LogisticRegression()
model.fit(X_train, y_train)

LogisticRegression()

### **Langkah 6: Menampilkan Parameter Regresi**
Mengeluarkan angka intercept (a) dan koefisien (b) dimaan pada regresi logistik, langkah ini digunakan untuk melihat seberapa besar pengaruh Usia dan Gaji terhadap keputusan pembelian dalam bentuk angka logaritma.

In [6]:
print("Intercept (a):", model.intercept_)
print("Koefisien (b1, b2):", model.coef_)
# b1 -> pengaruh Usia
# b2 -> pengaruh Gaji

Intercept (a): [-22.68608051]
Koefisien (b1, b2): [[0.12838338 0.27083911]]


### Analisis Hasil
*   **Intercept (a = -22.68) :** Angka ini menunjukkan titik awal peluang dalam skala logaritma. Nilai negatif yang cukup besar menandakan bahwa jika seseorang memiliki usia dan gaji yang sangat rendah (nol), maka peluang mereka untuk membeli mobil hampir tidak ada.
*  **Koefisien (b1 = 0.128) :** Setiap bertambahnya usia, kecenderungan atau peluang seseorang untuk membeli mobil akan meningkat. Karena nilainya positif, dapat disimpulkan bahwa semakin tua seseorang, semakin besar kemungkinan mereka memutuskan untuk membeli mobil
*  **Koefisien (b2 = 0.270) :** Sama seperti usia, faktor gaji juga memiliki pengaruh positif. Namun, perhatikan bahwa angka 0.270 lebih besar daripada 0.128. Ini berarti faktor Gaji memiliki pengaruh dua kali lebih kuat dibandingkan faktor Usia dalam menentukan keputusan pembelian mobil pada dataset ini.




### **Langkah 7: Prediksi Data Testing**
Proses ini menghasilkan output berupa label kategori (0 atau 1). Perbandingan antara hasil prediksi dan data aktual memberikan indikasi awal mengenai seberapa akurat model dalam membedakan konsumen potensial.

In [7]:
y_pred = model.predict(X_test)

print("Hasil Prediksi:", y_pred)
print("Data Sebenarnya:", y_test.values)

Hasil Prediksi: [1 1 0]
Data Sebenarnya: [0 1 0]


### Analisis Hasil
Dari 3 data uji yang ditampilkan, model berhasil menebak 2 data dengan benar dan salah pada 1 data. Ini menunjukkan akurasi sementara sebesar 66,6% pada sampel kecil ini.
#### Hasil Perbandingan:
*   Data ke-1: Prediksi menebak 1 (Beli), padahal aslinya adalah 0 (Tidak Beli). Terjadi kesalahan prediksi (False Positive).
*   Data ke-2: Prediksi menebak 1 (Beli) dan aslinya memang 1 (Beli). Tebakan Benar.
*   Data ke-3: Prediksi menebak 0 (Tidak Beli) dan aslinya memang 0 (Tidak Beli). Tebakan Benar.

Adanya kesalahan pada data pertama (menebak beli padahal tidak) menunjukkan bahwa model mungkin terlalu sensitif terhadap angka usia atau gaji tertentu, sehingga menganggap orang tersebut akan membeli padahal kenyataannya tidak.




### **Kesimpulan**
Implementasi regresi logistik pada data pembelian mobil ini membuktikan bahwa faktor Gaji punya pengaruh dua kali lebih kuat dibandingkan faktor Usia dalam menentukan seseorang jadi membeli mobil atau tidak. Lewat sistem klasifikasi ini, komputer bisa belajar membedakan kelompok "pembeli" dan "bukan pembeli" secara otomatis, di mana semakin mapan pendapatan dan usia seseorang, semakin besar peluang mereka untuk membeli. Meski masih ada sedikit tebakan yang meleset pada data uji, model ini sudah sangat berguna bagi tim marketing untuk mengincar calon pembeli yang paling potensial secara cepat dan akurat.